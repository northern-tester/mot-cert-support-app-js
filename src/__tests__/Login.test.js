import Login from '../components/Login';
import { render, fireEvent, waitFor } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import nock from 'nock';

const authMock = nock('http://localhost')
    .post('/v1/auth/login', {
        "email":"admin@test.com",
        "password":"password123"
    })
    .reply(200);

describe('Login component', () => {

    it('renders correctly', () => {
        const {asFragment} = render(<MemoryRouter>
                <Login />
        </MemoryRouter>);
        
        expect(asFragment()).toMatchSnapshot();
    });

    it('sends a login request correctly', async () => {
        const {getByTestId} = render(<MemoryRouter>
                <Login />
        </MemoryRouter>);

        fireEvent.change(getByTestId(/email/), { target: { value: 'admin@test.com' } });
        fireEvent.change(getByTestId(/password/), { target: { value: 'password123' } });
        await fireEvent.click(getByTestId("sign-in"));

        await waitFor(() => expect(authMock.isDone()).toBeTruthy());
    });

});