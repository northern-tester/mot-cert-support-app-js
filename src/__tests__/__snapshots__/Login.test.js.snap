// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 mt-5 mx-auto"
      >
        <form
          novalidate=""
        >
          <h1
            class="h3 mb-3 font-weight-normal"
          >
            Please sign in
          </h1>
          <div
            class="form-group"
          >
            <label
              for="email"
            >
              Email Address
            </label>
            <input
              class="form-control"
              data-testid="email"
              name="email"
              placeholder="Enter Email"
              type="email"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              for="password"
            >
              Password
            </label>
            <input
              class="form-control"
              data-testid="password"
              name="password"
              placeholder="Enter Password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-lg btn-primary btn-block mt-2"
            data-testid="sign-in"
            type="button"
          >
            Sign in
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
